packages/framework/src/resources/workflow/index.ts

```diff
@@ -81,6 +82,9 @@ export function workflow<
     };
   };
 
+  // TODO: delegate preference building to API later on. Must keep the mapping layer here though
+  const preferences = buildWorkflowChannelPreferences(mapPreferences(options.preferences));
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** As discussed with @rifont, we will later delegate any / all building of preferences to the API by removing this `buildWorkflowChannelPreferences`. This will require type changes in `framework` that will subsequently change types in `api` as well, so this will come at a later time‏